xdsl/dialects/func.py

```diff
@@ -69,6 +69,74 @@ def from_region(name: str, input_types: List[Attribute],
         op = FuncOp.build(attributes=attributes, regions=[region])
         return op
 
+    def replace_argument_type(self, arg: int | BlockArgument,
+                              new_type: Attribute):
+        """
+        Replaces the type of the argument specified by arg (either the index of the arg,
+        or the BlockArgument object itself) with new_type. This also takes care of updating
+        the function_type attribute.
+        """
+
+        if isinstance(arg, int):
+            if arg < 0:
+                arg = len(self.args) + arg
+            if not (0 <= arg < len(self.body.blocks[0].args)):
+                raise IndexError("Block {} does not have argument #{}!".format(
+                    self.body.blocks[0], arg))
+            arg = self.body.blocks[0].args[arg]
```

Comment:
Why are you doing all those checks here? AFAIU, the `Block`'s argument list is handling negative indexing

Reply:
uhh, actually yes! Let's remove this!